File: URI_1022_TDA_rational.js
const lines = [
    '4', 
    '1 / 2 + 3 / 4',
    '1 / 2 - 3 / 4', 
    '2 / 3 * 6 / 6',
    '1 / 2 / 3 / 4'
]

const N = Number(lines[0]);
for (let i = 0; i < N; i++) {
    const line = lines[i + 1].split(' ');

    const n1 = Number(line[0]);
    const d1 = Number(line[2]);
    const operator = line[3];
    const n2 = Number(line[4]);
    const d2 = Number(line[6]);

    let n = 0, d = 0;
    switch(operator){
        case '+':
            n = n1 * d2 + n2 * d1;
            d = d1 * d2;
            break;
        case '-':
            n = n1 * d2 - n2 * d1;
            d = d1 * d2;
            break;
        case '*':
            n = n1 * n2;
            d = d1 * d2;
            break;
        case '/':
            n = n1 * d2;
            d = d1 * n2;
            break;
    }

    let nReduced = 0, dReduced = 0;
    let initMcd = Math.abs(n);
    if (Math.abs(d) < initMcd) {
        initMcd = Math.abs(d);
    }

    for (let mcd = initMcd; mcd >= 2; mcd--) {
        if (n % mcd == 0 && d % mcd == 0) {
            nReduced = n / mcd;
            dReduced = d / mcd;
            break;
        }
    }	

    if (nReduced == 0) {
        nReduced = n;
        dReduced = d;
    }
    console.log(`${n}/${d} = ${nReduced}/${dReduced}`);
}

File: URI_1022_TDA_rational.java
import java.util.Scanner;

public class URI_1022_TDA_rational {

	public static void main(String[] args) {
		var sc = new Scanner(System.in);

		while(sc.hasNext()) {
			final short N = sc.nextShort();
			for (int i = 0; i < N; i++) {
				int n1 = sc.nextInt();
				sc.next();
				int d1 = sc.nextInt();
				String operator = sc.next();
				int n2 = sc.nextInt();
				sc.next();
				int d2 = sc.nextInt();

				int n = 0, d = 0;
				switch(operator){
					case "+":
						n = n1 * d2 + n2 * d1;
						d = d1 * d2;
						break;
					case "-":
						n = n1 * d2 - n2 * d1;
						d = d1 * d2;
						break;
					case "*":
						n = n1 * n2;
						d = d1 * d2;
						break;
					case "/":
						n = n1 * d2;
						d = d1 * n2;
						break;
				}

				int nReduced = 0, dReduced = 0;
				int initMcd = Math.abs(n);
				if (Math.abs(d) < initMcd) {
					initMcd = Math.abs(d);
				}

				for (int mcd = initMcd; mcd >= 2; mcd--) {
					if (n % mcd == 0 && d % mcd == 0) {
						nReduced = n / mcd;
						dReduced = d / mcd;
						break;
					}
				}	

				if (nReduced == 0) {
					nReduced = n;
					dReduced = d;
				}
				System.out.println(String.format("%d/%d = %d/%d", n, d, nReduced, dReduced));
			}
		}
		sc.close();
	}
}